Google.PowerShell/BigQuery/GcbqDataset.cs

```diff
@@ -0,0 +1,141 @@
+    ///   <para>This command will build a new dataset with the supplied datasetId, name, description, and a timeout of 1 day.</para>
+    /// </example>
+    /// <para type="link" uri="(https://cloud.google.com/bigquery/docs/reference/rest/v2/datasets)">
+    /// [BigQuery Datasets]
+    /// </para>
+    /// </summary>
+    [Cmdlet(VerbsCommon.New, "GcbqDataset")]
+    public class NewGcbqDataset : GcbqCmdlet
+    {
+        private class ParameterSetNames
+        {
+            public const string ByObject = "ByObject";
+            public const string ByValues = "ByValue";
+        }
+
+        /// <summary>
+        /// <para type="description">
+        /// The project to look for datasets in. If not set via PowerShell parameter processing, will
+        /// default to the Cloud SDK's DefaultProject property.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByObject)]
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByValues)]
+        [ConfigPropertyName(CloudSdkSettings.CommonProperties.Project)]
+        public string Project { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The dataset object that will be sent to the server to be inserted.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipeline = true, ParameterSetName = ParameterSetNames.ByObject)]
+        public Dataset Dataset { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The DatasetId for the datset reference object. Must be unique within the project, and must be 1024 
+        /// characters or less. It must only contain letters (a-z, A-Z), numbers (0-9), or underscores (_).
+        /// </para>
+        /// </summary>
+        [Parameter(Position = 0, Mandatory = true, ParameterSetName = ParameterSetNames.ByValues)]
+        [ValidatePattern("[a-zA-Z0-9]")]
+        [ValidateLength(1, 1024)]
+        public string DatasetId { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The "friendly name" field for the dataset. Used for a descriptive name for the dataset.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByValues)]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The "description" field for the dataset. Used for a user-friendly description of the dataset.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByValues)]
+        public string Description { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The default lifetime for tables in the dataset to exist in ms. Minimum is 3600000 (1hr).  
```

Comment:
Please grep through PowerShell docs (especially our other cmdlets) and make this consistent with the majority.

I believe most (all?) cmdlets use "seconds" as the unit of timeouts. MS isn't especially useful, and even if that is the implementation detail, we can hide that for our users.

(This is OK, since the output object says "ExpirationMs", so it won't lead to confusion.)